modules/openapi-generator-gradle-plugin/src/test/kotlin/GenerateTaskFromCacheTest.kt

```diff
@@ -0,0 +1,193 @@
+        configFile.createNewFile()
+        configFile.writeText("""{"foo":"bar"}""")
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `configFile - same directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeConfigFileTest()
+        runCacheabilityTestUsingSameDirectory(gradleVersion, configFileExtensionContents)
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `configFile - different directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeConfigFileTest()
+        runCacheabilityTestUsingDifferentDirectories(gradleVersion, configFileExtensionContents)
+    }
+
+    // ignoreFileOverride tests
+
+    private val ignoreFileOverrideExtensionContents = """
+        generatorName = "kotlin"
+        inputSpec = file("spec.yaml").absolutePath
+        ignoreFileOverride = file(".openapi-generator-ignore").absolutePath
+        """.trimIndent()
+
+    private fun initializeIgnoreFileTest() {
+        projectDir1.resolve(".openapi-generator-ignore").createNewFile()
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `ignoreFileOverride - same directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeIgnoreFileTest()
+        runCacheabilityTestUsingSameDirectory(gradleVersion, ignoreFileOverrideExtensionContents)
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `ignoreFileOverride - different directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeIgnoreFileTest()
+        runCacheabilityTestUsingDifferentDirectories(gradleVersion, ignoreFileOverrideExtensionContents)
+    }
+
+    // Helper methods & test fixtures
+
+    private fun runCacheabilityTestUsingSameDirectory(gradleVersion: String, extensionContents: String) {
+        // Arrange
+        withProject(extensionContents)
+
+        // Act
+        val result1 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        val result2 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "clean", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        // Assert
+        assertEquals(TaskOutcome.SUCCESS, result1.task(":openApiGenerate")?.outcome)
+        assertEquals(TaskOutcome.FROM_CACHE, result2.task(":openApiGenerate")?.outcome)
+    }
+
+    private fun runCacheabilityTestUsingDifferentDirectories(gradleVersion: String, extensionContents: String) {
+        // Arrange
+        withProject(extensionContents)
+        projectDir1.copyRecursively(projectDir2)
+
+        // Act
+        val result1 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "openApiGenerate")
```

Comment:
I would always use "clean" on tests where the expectation is that the outputs are cleaned, even when you can reasonably assume that the outputs don't exist (such as in this case).  First, it makes that expectation explicit to anyone reading the test.  Second, it future proofs the test against a potential change where another build is inserted before this build.

Reply:
Addressed by e355940d4c30e434f7c742c97a23347779bf9f7c